### PR TITLE
Fix docs of vcpu constructors

### PR DESCRIPTION
**Pull request by talhof8**
# Reason for This PR

Making the docs up-to-date.

## Description of Changes

Updating & fixing the documentation of both `Vcpu::new()` and `KvmVcpu::new()` which seem to be outdated. Also, removed a legacy description of a `vm_fd` argument.

- [X] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [X] All commits in this PR are signed (`git commit -s`).
- [X] The reason for this PR is clearly provided (issue no. or explanation).
- [X] The description of changes is clear and encompassing.
- [X] Any required documentation changes (code and docs) are included in this PR.
- [X] Any newly added `unsafe` code is properly documented.
- [X] Any API changes are reflected in `firecracker/swagger.yaml`.
- [X] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [X] All added/changed functionality is tested.

### CODE REVIEW

File: src/vmm/src/vstate/vcpu/mod.rs [hunk 1]
```diff
@@ -206,9 +206,8 @@ impl Vcpu {
     ///
     /// # Arguments
     ///
-    /// * `id` - Represents the CPU number between [0, max vcpus).
-    /// * `vm_fd` - The kvm `VmFd` for the virtual machine this vcpu will get attached to.
-    /// * `msr_list` - The `MsrList` listing the supported MSRs for this vcpu.
+    /// * `number` - Represents the CPU number between [0, max vcpus].
```
Comment (serban300) [inline src/vmm/src/vstate/vcpu/mod.rs:209]: Looks like the variable name is `index` here as well.

**Comment by dianpopa** [comment]
@talhof8 thanks for the contribution! There are some style related tests failing. Seems you need to use sign your commit with your name (i.e have a "signed off-by" line). You can use `git commit --amend -s`. I also suggest squashing the two commits into one.

File: src/vmm/src/vstate/vcpu/aarch64.rs [hunk 2]
```diff
@@ -68,7 +68,7 @@ impl KvmVcpu {
     ///
     /// # Arguments
     ///
-    /// * `id` - Represents the CPU number between [0, max vcpus).
+    /// * `index` - Represents the CPU number between [0, max vcpus].
```
Comment (luminitavoicu) [inline src/vmm/src/vstate/vcpu/aarch64.rs:71]: Would you mind removing the description for the `fd` argument on line 93, as well? It seems like arguments description for `configure()` function are out of date, too.

**Comment by talhof8** [comment]
Thank you for the CR!

All done, let me know if I'm missing anything :)

File: src/vmm/src/vstate/vcpu/aarch64.rs (hunk 2 above)
Comment (acatangiu) [reply src/vmm/src/vstate/vcpu/aarch64.rs:71]: It should be `[0, max_vcpus)` or if you find that notation confusing, then use `0 <= index < max_vcpus`.

`index` will be strictly smaller than `max_vcpus`.

File: src/vmm/src/vstate/vcpu/x86_64.rs [hunk 3]
```diff
@@ -169,7 +169,7 @@ impl KvmVcpu {
     ///
     /// # Arguments
     ///
-    /// * `id` - Represents the CPU number between [0, max vcpus).
+    /// * `index` - Represents the CPU number between [0, max vcpus].
```
Comment (acatangiu) [inline src/vmm/src/vstate/vcpu/x86_64.rs:172]:
```suggestion
    /// * `index` - Represents the 0-based CPU index.
```

**Review by acatangiu — COMMENTED** [review]
Looks good, please squash* your commits and this PR should be good to go.

* https://www.git-tower.com/learn/git/faq/git-squash/

**Review by acatangiu — APPROVED** [review]

**Review by luminitavoicu — APPROVED** [review]

**Pull request merged**